src/aihwkit/inference/noise/pcm.py

```diff
@@ -102,6 +108,34 @@ def __init__(
         self.prog_noise_scale = prog_noise_scale
         self.read_noise_scale = read_noise_scale
         self.drift_scale = drift_scale
+        self.valid_kwargs = ['custom_drift_model']
+
+        if not all(key in self.valid_kwargs for key in kwargs):
+            ValueError("PCMLikeNoiseModel only supports kwargs = %s" % self.valid_kwargs)
+
+        self.custom_drift_model = kwargs.get('custom_drift_model')
+
+        if self.custom_drift_model is not None:
+            drift_model_g_min = min(self.custom_drift_model['g_lst'])
```

Comment:
These hidden args are bad code practice. Why not give these explicitly in a new class ?